### PR TITLE
Minimal flag

### PR DESCRIPTION
**Pull request by alan-agius4**
fix(@schematics/angular): minimal should be honored in workspace creation
refactor: clean up code for minimal flag

### CODE REVIEW

File: packages/schematics/angular/application/index.ts [hunk 1]
```diff
@@ -261,20 +260,20 @@ export default function (options: ApplicationOptions): Rule {
...
+      {
+        inlineStyle: options.inlineStyle,
+        inlineTemplate: options.inlineTemplate,
+        spec: !options.skipTests,
+        styleext: options.style,
+        viewEncapsulation: options.viewEncapsulation,
+      } :
+      {
+        inlineStyle: true,
+        inlineTemplate: true,
```
Comment (alan-agius4) [inline packages/schematics/angular/application/index.ts]: oops there was a typo here. Let's be more strict when it comes to types.

File: packages/schematics/angular/application/index_spec.ts [hunk 2]
```diff
@@ -146,6 +146,30 @@ describe('Application Schematic', () => {
     expect(confContent.projects['foo-e2e']).toBeUndefined();
   });
 
+  it('should create correct files when using minimal', () => {
+    const options = { ...defaultOptions, minimal: true };
+    const tree = schematicRunner.runSchematic('application', options, workspaceTree);
+    const files = tree.files;
+
+    expect(files.indexOf('/projects/foo/karma.conf.js')).toBe(-1);
```
Comment (hansl) [inline packages/schematics/angular/application/index_spec.ts]: Please use `toContain()` or `.not.toContain()` instead of looking for indices...

**Comment by vikerman** [comment]
@alan-agius4 - Does not merge cleanly into 7.0.x branch. Please send PR to 7.0.x branch separately

File: packages/schematics/angular/application/index_spec.ts (hunk 2 above)
Comment (alan-agius4) [reply packages/schematics/angular/application/index_spec.ts]: @hansl sry about that, I'll update this and a couple of them in a seperate PR.

**Pull request merged**